.github/workflows/ci.yaml

```diff
@@ -15,21 +15,82 @@ env:
   NODE_OPTIONS: --max_old_space_size=6144
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+permissions: {}
+
 jobs:
+  dedupe:
+    name: Deduplicate dependencies
+    # Skip unless this is a dependabot pull request
+    if: |
+      github.actor == 'dependabot[bot]' &&
+      startsWith(github.head_ref, 'dependabot/npm_and_yarn/')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    outputs:
+      # Downstream jobs need to use this SHA to get the dedupe commit
+      sha: ${{ steps.get-sha.outputs.sha }}
+    steps:
+      - name: Check out files from GitHub
+        uses: actions/checkout@v3
+        with:
+          # Checkout PR head instead of merge commit
+          # Use ref, not SHA, so reruns get the dedupe commit
+          ref: ${{ github.event.pull_request.head.ref }}
+      - name: Set up Node ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: yarn
+      - name: Install dependencies
+        # Do not run build scripts as a security measure since job has write permissions
+        run: yarn install --immutable --mode=skip-build
+      - name: Deduplicate dependencies
+        run: yarn dedupe --mode=skip-build
+      - name: Commit changes
+        run: |
+          git config user.name "GitHub Action"
+          git config user.email "github-action@users.noreply.github.com"
+          git add yarn.lock
+          git commit -m "Deduplicate dependencies" || exit 0
+          git push origin HEAD:$GITHUB_HEAD_REF
+          echo "DEDUPED=true" >> $GITHUB_ENV
+      - name: Output updated SHA for merge commit
+        id: get-sha
+        shell: bash
+        timeout-minutes: 15
+        run: |
+          if [ -v DEDUPED ]; then
+            echo "Waiting for GitHub to do the mergability check and update the commit SHA..."
+            while [ -z "$sha" -o "$sha" == "$GITHUB_SHA" ]; do
+              sleep 5s
+              sha=`git ls-remote origin $GITHUB_REF | awk '{print $1}'`
+            done
```

Comment:
```suggestion
            while [[ -z "${sha}" -o "${sha}" == "${GITHUB_SHA}" ]; do
              sleep 5s
              sha=$(git ls-remote origin "${GITHUB_REF}" | awk '{print $1}')
            done
```

Reply:
Can't use `-o` inside `[[`.  Rather than risk errors, I just ran the bash through a test first.  See new commit.